app/src/main/java/me/guillaumin/android/osmtracker/db/DataHelper.java

```diff
@@ -65,6 +68,9 @@
 	 */
 	private Context context;
 
+	/** Statistics for all existing tracks */
+	private static TreeMap<Long, TrackStatistics> tracksStatistics = new TreeMap<Long, TrackStatistics> ();
```

Comment:
I can see it only in 2 classes (Assuming the DataHelper one is removed)?

I would create a field in `TrackDetail` which I think makes sense because the `TrackDetail` activity is about a specific track, so having a field holding its data is reasonable and useful as it needs to be updated. You can create it in `onStart()` and then refer to it / update it in the various methods like `onResume()`.

In `TracklistAdapter` it's a bit tricky since there are the need for one of them for each track. I guess you could use the `Map` idea here perhaps... 

The problem with storing it in `DataHelper` is that it's not the idiomatic way to share data between activities and views in Android and that's likely to cause issues and bug difficult to track down from past experience. You'll also need to deal with events like tracks being deleted (removing items from the map). It also complicates maintenance because if we have something else updating the DB not going through `DataHelper.track()` then the track statistics won't be updated

It would be preferable for the data to be local to the component that needs it (Activity, View) and to tie it to the lifecycle of this component (the `TrackStatstics` object gets garbage collected when the view is removed)